---
title: "Exposure MS on outcome MDD"
output: html_notebook
author: "Mitul Patel"
date: "`r format(Sys.time(), '%d %B %Y')`"
---

##### The package used for ***TwoSampleMR analysis***: [https://mrcieu.github.io/TwoSampleMR/]. Here we use TwoSampleMR (2SMR) to estimate the causal effect of an exposure (MS) on an outcome (depression). **Steps to perform**:
 1. Select instruments for the exposure (i.e MS data with pval significant SNPs and LD clumped).
 2. Select outcome (i.e depression data)
 3. Harmonise the effect sizes for the instruments on the exposure and the outcomes to be each for the same reference allele. If different allele, the effect sizes can be seem to have a negative sign. 
 4. Perform MR analysis
 
### Exposure
```{r}
library(TwoSampleMR)
library(dplyr)
library(data.table)
library(tidyr)
library(ieugwasr)

# Minimum information needed in the instruments for an exposure: SNP, beta (logOR for binary case: logOR == bea == logEffectSize), se, effect_allele. The significant snps from MS dataset are extracted and LD clumped using clump_data(). LD clump ensures that the instruments for the exposure are independent. (add the size of the original data and the source of the data)

ms_dat <- fread("../data/MS_data2020.csv") %>% as_tibble() 
dim(ms_dat)
ms_dat <- ms_dat %>% filter(P < 5.0e-8)
ms_dat <- ms_dat %>% select(ID, CHR, A1, A2, P, OR, A1_Freq, beta, logEffectSize, logStandardError, logStandardDeviation, LLCI, ULCI, EffectSize) %>% rename(SNP = ID, se = logStandardError, effect_allele = A1, other_allele = A2, pval = P, eaf = A1_Freq) %>% mutate(exposure = "ms")
dim(ms_dat)
ms_dat <- clump_data(ms_dat)
ms_dat<- format_data(ms_dat, type ="exposure") # to get the data in the format that MRBase accepts
dim(ms_dat)
head(ms_dat)
```

### Outcome
```{r}
# Extract the MS SNPs from the MDD: (add the source of the data)
depression_dat <- fread("../data/PGC_UKB_depression_genome-wide.fix.tsv",sep='\t', header = TRUE) %>% as_tibble() %>% rename(SNP = MarkerName, beta = LogOR, se = StdErrLogOR, effect_allele = A1, other_allele = A2, pval = P, eaf = Freq) %>% mutate(outcome = "depression")
# write.csv(depression_dat, "../data/MDD2020.csv")
dim(depression_dat)
head(depression_dat)

# overlapping snps
ms_dep_snps <- inner_join(ms_dat, depression_dat, by="SNP")
dim(ms_dep_snps)
head(ms_dep_snps)
```

```{r}
# Of the 81 SNPs in the exposure data, 76 SNPs were present in the ms data. The remaining 5 needs can be found via LD proxies.
SNPs_not_persent <- subset(ms_dat, !(SNP %in% ms_dep_snps$SNP))
SNPs_not_persent
```
```{r}
# Alternate option to extract the depression SNPs from MS using the function from MRBase: found 45 again (verifies above results!)
outcome_dat <- read_outcome_data(
  ms_dat$SNP,
  filename = "../data/MDD2020.CSV",
  sep=",",
  snp_col = "SNP",
  beta_col = "beta",
  se_col = "se",
  effect_allele_col = "effect_allele",
  other_allele_col = "other_allele",
  eaf_col = "eaf",
  pval_col = "pval"
)
dim(outcome_dat)
head(outcome_dat)
```


```{r}
# In total 151 proxies were found. Here we select a proxy for each leadSNP that has the smalles p-val in P column and is stored in the "20200114_mdd_ms_proxies.csv" file. 
proxy_sig <- fread("../data/20200126_proxy_MSexp.csv", header = T) %>% as_tibble() %>% select(CHROM, ID, REF, ALT, MAF, leadSNP, POS, R.squared) %>% rename(SNP=leadSNP,effect_allele=REF, other_allele=ALT, Freq=MAF, proxySNP = ID) # rename so its inline with the other 45 SNPs data
head(proxy_sig)
proxy_sig <- inner_join(ms_dat, proxy_sig, by="SNP") 
proxy_sig <- proxy_sig[!duplicated(proxy_sig$SNP),]

## JOIN BY WHICH COLUMNS
proxy_sig_dat <- left_join(proxy_sig, depression_dat, by="SNP")
proxy_sig_dat <- format_data(proxy_sig, type="outcome", snps = NULL, header = T, snp_col = "SNP", beta_col = "beta.exposure", se_col = "se.exposure", effect_allele_col = "effect_allele.exposure", other_allele_col = "other_allele.exposure", pval_col = "pval.exposure",eaf_col = "eaf.exposure") # get the data in the right format for MRBase
proxy_sig_dat$id.outcome <- outcome_dat$id.outcome[1]
proxy_sig_dat <- proxy_sig_dat %>% mutate(data_source.outcome = 'textfile') # so the both datasets have right number of cols

#  merging 70 SNPs with the 5 LD proxies
proxysig_ms_dat <- rbind(proxy_sig_dat, outcome_dat) 
dim(proxysig_ms_dat)
```


### Harmonise
```{r}
# without the 5 proxied SNPs: 
harmonise_76snps_dat <- harmonise_data(exposure_dat = ms_dat,
                      outcome_dat = outcome_dat)
dim(harmonise_76snps_dat)
head(harmonise_76snps_dat)
```
```{r}
# with the 5 proxied SNPs
harmonise_dat <- harmonise_data(exposure_dat = ms_dat,
                      outcome_dat = proxysig_ms_dat)
dim(harmonise_dat)
head(harmonise_dat)
```

### MR
```{r}
res_no_proxy <- mr(harmonise_76snps_dat, method_list = c("mr_egger_regression", "mr_ivw"))
res_no_proxy

res <- mr(harmonise_dat, method_list = c("mr_egger_regression", "mr_ivw"))
res
write.csv(res,"../data/20200121_mr_expMS_outMDD.csv")
```

### Sensitivity analysis : Heterogeneity stats
```{r}
mr_heterogeneity(harmonise_dat, method_list=c("mr_egger_regression", "mr_ivw"))
mr_pleiotropy_test(harmonise_dat) # Horizontal pleiotropy: intercept term
```

### Leave-one-out
```{r}
res_loo <- mr_leaveoneout(harmonise_dat) # identify which SNP(s) is driving the association.
dim(res_loo)
```

### Plots
```{r}
res <- mr(harmonise_dat, method_list = c("mr_egger_regression","mr_ivw"))

# scatter plot
p1 <- mr_scatter_plot(res,harmonise_dat) # IVW line is the standard MR analysis, and the MR Egger is the pleiotropy-adjusted MR-Egger regression line. The estimated MR-Egger intercept term was 0.01.
p1[[1]]
#ggsave(p1[[1]], file="../results/scatter plot.pdf", width=7, height=7)

# forest plot
res_single <- mr_singlesnp(harmonise_dat, all_method=c("mr_ivw", "mr_two_sample_ml")) #
p2 <- mr_forest_plot(res_single)
p2[[1]]

# leave-one-out
res_loo <- mr_leaveoneout(harmonise_dat)
p3 <- mr_leaveoneout_plot(res_loo)
p3[[1]]

# funnel plot
res_single <- mr_singlesnp(harmonise_dat)
p4 <- mr_funnel_plot(res_single)
p4[[1]]
```

